dqc_us_rules/dqc_us_0005.py

```diff
@@ -44,36 +65,74 @@ def _get_end_of_period(val):
+
+    :param val: val to check
+    :type val: val
+    """
     eop_results = _get_end_of_period(val)
-    fact_dict = facts.LegalEntityAxis_facts_by_member(filter(lambda f: f.context is not None, val.modelXbrl.facts))
+    fact_dict = facts.legal_entity_axis_facts_by_member(
+        filter(lambda f: f.context is not None, val.modelXbrl.facts)
+    )
     for lea_member, fact_list in fact_dict.items():
-        lookup = lea_member if lea_member in eop_results else facts.LEGALENTITYAXIS_DEFAULT
+        lookup = (
+            lea_member
+            if lea_member in eop_results
+            else facts.LEGALENTITYAXIS_DEFAULT
+        )
         if lookup in eop_results:
             for fact in fact_list:
-                fact_date = fact.context.endDatetime  # endDateTime will be the instant date time if this is an instant period
+                # endDateTime will be the instant date time if this
+                # is an instant period
+                fact_date = fact.context.endDatetime
                 if fact_date is not None:
-                    # Check for the case where a fact is less than the expected eop dates
+                    # Check for the case where a fact is less than
+                    # the expected eop dates
                     comparison_date = eop_results[lookup][1]
                     if fact_date <= comparison_date:
-                        if fact.localName == 'EntityCommonStockSharesOutstanding':
-                            # if a fact whose qname is EntityCommonStockSharesOutstanding has an end date prior to eop then fire an error
-                            if fact_date < comparison_date:  # Only fire if it is actually less than the comparison date
-                                val.modelXbrl.error('{base_code}.17'.format(base_code=_CODE_NAME), messages.get_message(_CODE_NAME, "17"),
-                                                    modelObject=[fact] + list(eop_results[lookup]),
-                                                    ruleVersion=_RULE_VERSION)
-                        elif facts.axis_exists(val, fact, 'SubsequentEventTypeAxis'):
-                            val.modelXbrl.error('{base_code}.48'.format(base_code=_CODE_NAME), messages.get_message(_CODE_NAME, "48"),
-                                                modelObject=[fact] + list(eop_results[lookup]),
-                                                ruleVersion=_RULE_VERSION)
-                        elif facts.axis_member_exists(val, fact, 'StatementScenarioAxis', 'ScenarioForecastMember'):
-                            val.modelXbrl.error('{base_code}.49'.format(base_code=_CODE_NAME), messages.get_message(_CODE_NAME, "49"),
-                                                modelObject=[fact] + list(eop_results[lookup]),
-                                                ruleVersion=_RULE_VERSION)
+                        ECSSO = 'EntityCommonStockSharesOutstanding'
+                        if fact.localName == ECSSO:
```

Comment:
Lets cut a ticket out for extracting this entire if/elif/else block out to a helper function. This level of ugly formatting is a clear code smell.